iothub_client/samples/iothub_client_sample_module_sender/iothub_client_sample_module_sender.c

```diff
@@ -41,7 +41,7 @@ int main(void)
     IOTHUB_MODULE_CLIENT_LL_HANDLE iotHubModuleClientHandle;
     EVENT_INSTANCE messages[MESSAGE_COUNT];
 
-    srand((unsigned int)time(NULL));
+    srandom((unsigned int)time(NULL));
```

Comment:
Here and everywhere:
srand is a C99-spec function. It is supported by every platform that supports C99, as azure-iot-sdk-c is expected to support.

srandom is a POSIX-1 function.
https://linux.die.net/man/3/srandom

This means that with this change, not all platforms will be supported by the azure-iot-sdk-c. 
We might disrupt customers with this change.

I'm inclined to say that we will need an adapter in c-utility for this (like it has been done for other related problems). If the platform is Linux, then use srandom. Otherwise, use srand.